mtwitter.js

```diff
@@ -0,0 +1,244 @@
+    this.timeline_queue = [];
+    this.timeline_period = 0;
+    this.timeline_intervalobj = null;
+    this._bridge = bridge;
+    this.get_bearer_token = function(cb) {
+
+        try {
+            fs.accessSync('bearer.tok', fs.R_OK);
+            var tok = fs.readFileSync('bearer.tok', 'utf8');
+            cb(tok);
+            return;
+        } catch (e) {
+            console.log("Bearer token either does not exist or cannot be read, getting a new token.")
+        }
+
+        var key = this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret;
+        key = Buffer.from(this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret, 'ascii').toString('base64');
+        var options = {
+            url: "https://api.twitter.com/oauth2/token",
+            headers: {
+                'Authorization': "Basic " + key
+            },
+            form: "grant_type=client_credentials",
+            contentType: "application/x-www-form-urlencoded;charset=UTF-8"
+        };
+        Request.post(options, function(error, response, body) {
+            if (error) {
+                console.log("Error", error);
+                return false;
+            } else {
+                try {
+                    var jsonresponse = JSON.parse(body);
+                    if (jsonresponse.token_type == "bearer") {
+                        fs.writeFile("bearer.tok", jsonresponse.access_token, (err) => {
+                            if (err) {
+                                console.console.error("Couldn't write bearer token to file.");
+                            }
+                        });
+                        cb(jsonresponse.bearer_token);
+                    } else {
+                        console.error("Error getting bearer token: Unexpected response");
+                        cb(null);
+                    }
+                } catch (e) {
+                    console.error("Error getting bearer token:", e);
+                    cb(null);
+                }
+
+            }
+        });
+    }
+
+    this.get_bearer_token((bt) => {
+        if (bt != false) {
+            this.app_auth.bearer_token = bt;
+            console.log("Twitter Application Auth OK!");
+            this.app_twitter = new Twitter(this.app_auth);
+        } else {
+            console.error("Twitter Application Failed Auth. The bridge will be operating in a limited capacity.");
+        }
+    });
+
+    this.get_user_by_id = function(id) {
+        var ts = new Date().getTime();
+        return new Promise((resolve, reject) => {
+            for (var name in this.tuser_cache) {
+                var cached = this.tuser_cache[name];
+                if (ts - cached.cache_time > TWITTER_PROFILE_INTERVAL) {
+                    continue;
+                }
+                if (cached.user != null && cached.user.id_str == id) {
+                    resolve(cached.user);
+                }
+            }
+            var twit = this;
```

Comment:
If you're going to alias `this`, always use a consistent, generic term to avoid confusion (typically we use `self`). Though in practice, you're *already using arrow functions*, so I would remove this entirely and arrow function the closures.